Tests/GUI/Dialog/DMatLib/test_DMatSetup.py

```diff
@@ -609,11 +625,156 @@ def test_BH_setup(self):
+        assert wimport.b_ok.isEnabled() == True
+        wimport.b_ok.clicked.emit()
+
+        assert mat.si_row.value() == 50
+        assert mat.si_col.value() == 3
+        assert mat.w_tab.cellWidget(0, 0).value() == 50
+        assert mat.w_tab.cellWidget(1, 0).value() == 50
+        assert mat.w_tab.cellWidget(2, 0).value() == 50
+        assert mat.w_tab.cellWidget(3, 0).value() == 50
+        assert mat.w_tab.cellWidget(49, 0).value() == 100
+
+        assert mat.w_tab.cellWidget(0, 1).value() == 0
+        assert mat.w_tab.cellWidget(1, 1).value() == 1
+        assert mat.w_tab.cellWidget(2, 1).value() == 2
+        assert mat.w_tab.cellWidget(3, 1).value() == 3
+        assert mat.w_tab.cellWidget(49, 1).value() == 49
+
+        assert mat.w_tab.cellWidget(0, 2).value() == 0
+        assert mat.w_tab.cellWidget(1, 2).value() == 0.01
+        assert mat.w_tab.cellWidget(2, 2).value() == 0.02
+        assert mat.w_tab.cellWidget(3, 2).value() == 0.03
+        assert mat.w_tab.cellWidget(49, 2).value() == 0.49
+
+        assert self.widget.w_setup.mat.mag.LossData.value[1, 3] == 3
+        assert self.widget.w_setup.mat.mag.LossData.value[0, 3] == 50
+
+        
+
+        # Export Csv
+        save_csv_path_Loss = join(save_path, "DMatSetup_csv_export_Loss.csv").replace(
+            "\\", "/"
+        )
+        with mock.patch(
+            "PySide2.QtWidgets.QFileDialog.getSaveFileName",
+            return_value=(save_csv_path_Loss, "CSV (*.csv)"),
+        ):
+            mat.b_export.clicked.emit()
+        assert isfile(save_csv_path_Loss)
+
+        # Load csv
+        with mock.patch(
+            "PySide2.QtWidgets.QFileDialog.getOpenFileName",
+            return_value=(save_csv_path_Loss, "CSV (*.csv)"),
+        ):
+            mat.b_import.clicked.emit()
+
+        assert mat.si_row.value() == 50
+        assert mat.si_col.value() == 3
+        assert mat.w_tab.cellWidget(0, 0).value() == 50
+        assert mat.w_tab.cellWidget(1, 0).value() == 50
+        assert mat.w_tab.cellWidget(2, 0).value() == 50
+        assert mat.w_tab.cellWidget(3, 0).value() == 50
+        assert mat.w_tab.cellWidget(49, 0).value() == 100
+
+        assert mat.w_tab.cellWidget(0, 1).value() == 0
+        assert mat.w_tab.cellWidget(1, 1).value() == 1
+        assert mat.w_tab.cellWidget(2, 1).value() == 2
+        assert mat.w_tab.cellWidget(3, 1).value() == 3
+        assert mat.w_tab.cellWidget(49, 1).value() == 49
+
+        assert mat.w_tab.cellWidget(0, 2).value() == 0
+        assert mat.w_tab.cellWidget(1, 2).value() == 0.01
+        assert mat.w_tab.cellWidget(2, 2).value() == 0.02
+        assert mat.w_tab.cellWidget(3, 2).value() == 0.03
+        assert mat.w_tab.cellWidget(49, 2).value() == 0.49
+
+        assert self.widget.w_setup.mat.mag.LossData.value[1, 3] == 3
+        assert self.widget.w_setup.mat.mag.LossData.value[0, 3] == 50
+        assert self.widget.w_setup.mat.mag.LossData.value[2, 3] == 0.03
+
```

Comment:
add click on plot

Reply:
fig, ax to catch the plot
check that it is set, check figure/ax title + number lines/legend

Reply:
self.fig is None
plot
self.fig is not None 
Check axes content
self.fig.close()